src/riak_kv_qry_compiler.erl

```diff
@@ -95,26 +116,123 @@ now_milliseconds() ->
+%% it generates InvDistFuns in a foldl.
+compile_order_by(?SQL_SELECT{'SELECT' = Select,
+                             'WHERE'  = [Where]} = Q,
+                 InvDistFuns = [_|_]) ->
+    %% Offset is now a computable thing
+    case compile_inverdist_funcalls_to_orderby(InvDistFuns) of
+        {CompiledSpecs, []} ->
+            {CompiledOrderBys,
+             CompiledLimits,
+             CompiledOffsets} = lists:unzip3(CompiledSpecs),
+            case check_invdist_funs_consistent_with_select(
+                   CompiledOrderBys, Select) of
+                ok ->
+                    %% Select has one compiled column per PERCENTILE call, but
+                    %% only one makes sense
+                    [{SubstituteColName, _, _}|_] = CompiledOrderBys,
+                    %% add a WHERE $that_column is not null because
+                    %% inverse distribution functions want nulls
+                    %% discarded
+                    ExtendedWhere = {and_, {is_not_null, {identifier, SubstituteColName}},
+                                     Where},
+                    SingleColumnSelect = make_single_column_select(Select, SubstituteColName),
+                    {ok, Q?SQL_SELECT{'SELECT'   = SingleColumnSelect,
+                                      'ORDER BY' = CompiledOrderBys,
+                                      'OFFSET'   = CompiledOffsets,
+                                      'LIMIT'    = CompiledLimits,
+                                      'WHERE'    = [ExtendedWhere]}};
+                ER ->
+                    ER
+            end;
+        {_, BadFunCalls} ->
+            %% there can be multiple errors to report, but we can only
+            %% report one, so:
+            {error, hd(BadFunCalls)}
+    end;
+
+%% this is the real ORDER BY (no InvDistFuns)
+compile_order_by(?SQL_SELECT{'ORDER BY' = OrderBy} = Q,
+                 _InvDistFuns = []) ->
     case non_unique_identifiers(OrderBy) of
         [] ->
-            {ok, Q?SQL_SELECT{'ORDER BY' = OrderBy}};
+            {ok, Q};
         WhichNonUnique ->
-            {error, {non_unique_orderby_fields, ?E_NON_UNIQUE_ORDERBY_FIELDS(hd(WhichNonUnique))}}
+            {error, {non_unique_orderby_fields,
+                     ?E_NON_UNIQUE_ORDERBY_FIELDS(hd(WhichNonUnique))}}
     end.
 
+
+
+-spec check_invdist_funs_consistent_with_select(list(tuple()), #riak_sel_clause_v1{}) ->
+                                                       ok | {error, tuple()}.
+check_invdist_funs_consistent_with_select(CompiledOrderBys,
+                                          #riak_sel_clause_v1{col_names = ColNames}) ->
+    lists:foldl(
+      fun(F, ok) -> F(); (_F, Error) -> Error end,
+      ok,
+      [fun() ->
+               %% we allow multiple PERCENTILE calls in a single SELECT,
+               %% but they all must have the same column argument
+               case lists:usort(CompiledOrderBys) of
+                   [_SameColumndOrderBy] ->
```

Comment:
Nit, but it might be important to understand the case clause: there's an apparently-extraneous `d` in the binding name, `_SameColumndOrderBy`